vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -1,18 +1,22 @@
 #!/bin/bash -x
 
+if [[ ${SUBSCRIPTION_ID} == ${PROD_SUBSCRIPTION_ID} ]]; then
```

Comment:
Does this expect that Linux does not set `PROD_SUBSCRIPTION_ID` so PROD_SUBSCRIPTION_ID is empty and if is false?
Suggest to check `windowsVhdMode` before others.

Reply:
Currently backfill cleanup is only used by Windows pipeline. I could check `windowsVhdMode` to avoid such issue.